core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManager.java

```diff
@@ -401,8 +450,13 @@ public InputStream fetchIndex(final RemoteLogSegmentMetadata remoteLogSegmentMet
 
             final var segmentManifest = fetchSegmentManifest(remoteLogSegmentMetadata);
 
-            final var key = objectKey(remoteLogSegmentMetadata, ObjectKeyFactory.Suffix.fromIndexType(indexType));
-            final var in = fetcher.fetch(key);
+            final var key = objectKey(remoteLogSegmentMetadata, ObjectKeyFactory.Suffix.INDEXES);
+            final var segmentIndex = segmentManifest.segmentIndexes().segmentIndex(indexType);
+            if (segmentIndex == null) {
+                throw new RemoteResourceNotFoundException("Index " + indexType + " not found on " + key);
+            }
+            final var range = BytesRange.ofFromPositionAndSize(segmentIndex.position(), segmentIndex.size());
```

Comment:
We can probably add `segmentIndex.range()` or something of this sort, but your call :)

Reply:
Sure, good idea. Will add it.